### ISSUE TITLE
Allow using env variables in confmap.Provider URI

### ISSUE THREAD

**Issue by tigrannajaryan** [open-telemetry/opentelemetry-collector#7176]
## Problem Description

It is currently impossible to use env variable in confmap.Provider.Resolve() URIs. Trying something like this:

```yaml
receivers: ${file:${HOME}myreceivers.yaml}
```

Results in an error message: `Error: failed to get config: cannot resolve the configuration: the uri "file:${HOME" contains unsupported characters ('$')
2023/02/10 16:04:08 collector server run finished with error: failed to get config: cannot resolve the configuration: the uri "file:${HOME" contains unsupported characters ('$')`

## Proposed Solution

Allowing using env variables in the URL of confmap.Provider. 

This will enable interesting use cases. For example assuming [OTELCOL_CFG_LOCATION_BASE](https://github.com/open-telemetry/opentelemetry-collector/issues/7175) proposal is implemented we can reference config files relative to each other like this.

```yaml
# this is the main config file.

# Include other config files from the same directory where the main config file is located.
receivers: ${file:${OTELCOL_CFG_LOCATION_BASE}receivers.yaml}
processors: ${file:${OTELCOL_CFG_LOCATION_BASE}processors.yaml}
exporters: ${file:${OTELCOL_CFG_LOCATION_BASE}exporters.yaml}
```

**Comment by mx-psi** [comment]
I think this is a duplicate of #7117

**Comment by tigrannajaryan** [comment]
https://github.com/open-telemetry/opentelemetry-collector/issues/7117 also says "For the moment probably we should document that this is not supported and forbidden". 

If that's the conclusion then let's keep this one open to be done after #7117 is closed with documenting and forbidding.

**Comment by bogdandrutu** [comment]
It is still duplicate, just that my propose solution is not what you expect, which is fine and I am happy to have someone implementing that.

**Comment by tigrannajaryan** [comment]
> It is still duplicate, just that my propose solution is not what you expect, which is fine and I am happy to have someone implementing that.

Sure, feel free to close this one if you think https://github.com/open-telemetry/opentelemetry-collector/issues/7117 captures all that is needed.

**Comment by mx-psi** [comment]
Resolved by #7504
